ArcGISRuntimeSDKQt_CppSamples/Routing/OfflineRouting/OfflineRouting.cpp

```diff
@@ -0,0 +1,271 @@
+    {
+      m_selectedGraphic = nullptr;
+      // find corresponding graphic in m_stopsOverlay
+      for (int i = 0; i < m_stopsOverlay->graphics()->size(); ++i)
+      {
+        if (m_stopsOverlay->graphics()->at(i)->operator==(result->graphics().at(0)))
+        {
+          m_selectedGraphic = m_stopsOverlay->graphics()->at(i);
+          break;
+        }
+      }
+    }
+
+  });
+
+  // check whether mouse pressed over an existing stop
+  connect(m_mapView, &MapQuickView::mousePressed, this, [this](QMouseEvent& e){
+    m_mapView->identifyGraphicsOverlay(m_stopsOverlay, e.x(), e.y(), 10, false);
+  });
+
+  // get stops from clicked locations
+  connect(m_mapView, &MapQuickView::mouseClicked, this, [this](QMouseEvent& e){
+    if (!m_selectedGraphic)
+    {
+      SimpleMarkerSymbol* stopLabel = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle::Circle, Qt::red, 20, this);
+      Graphic* stopGraphic = new Graphic(m_mapView->screenToLocation(e.x(), e.y()), stopLabel, this);
+      m_stopsOverlay->graphics()->append(stopGraphic);
+      findRoute();
+    }
+    e.accept();
+  });
+
+  connect(m_mapView, &MapQuickView::mouseReleased, this, [this](QMouseEvent& e) {
+    if (m_selectedGraphic)
+    {
+      m_selectedGraphic = nullptr;
+      e.accept();
+    }
+  });
+
+  connect(m_mapView, &MapQuickView::mouseMoved, this, [this](QMouseEvent&e){
+    if (m_selectedGraphic)
+    {
+      e.accept();
+      m_selectedGraphic->setGeometry(m_mapView->screenToLocation(e.x(), e.y()));
+      findRoute();
+    }
+  });
+
+  connect(m_routeTask, &RouteTask::solveRouteCompleted, this, [this](QUuid, const RouteResult routeResult){
+    // clear old route
+    m_routeOverlay->graphics()->clear();
+    Polyline routeGeometry = routeResult.routes().first().routeGeometry();
+    Graphic* routeGraphic = new Graphic(routeGeometry, this);
+
+    m_routeOverlay->graphics()->append(routeGraphic);
+  });
+}
+
+void OfflineRouting::findRoute()
+{
+  if(!m_taskWatcher.isDone()) {
+    return;
+  }
+
+  int numGraphics = m_stopsOverlay->graphics()->size();
+  if (numGraphics > 1)
+  {
+    QList<Stop> stops;
+    for (int i = 0; i < numGraphics; ++i)
+    {
+      stops.append(Stop(m_stopsOverlay->graphics()->at(i)->geometry()));
+    }
+
+    // configure stops and travel mode
+    m_routeParameters.setStops(stops);
+    m_routeParameters.setTravelMode(m_routeTask->routeTaskInfo().travelModes().at(m_travelModeIndex));
+
+    // solve the route
+    m_taskWatcher = m_routeTask->solveRoute(m_routeParameters);
+  }
+}
+
+// Set the view (created in QML)
+void OfflineRouting::setMapView(MapQuickView* mapView)
+{
+  if (!mapView || mapView == m_mapView)
+    return;
+
+  m_mapView = mapView;
+  m_mapView->setMap(m_map);
+
+  m_mapView->graphicsOverlays()->append(m_stopsOverlay);
+  m_mapView->graphicsOverlays()->append(m_routeOverlay);
+
+  const QString geodatabaseLocation = QString("%1/ArcGIS/Runtime/Data/tpk/san_diego/sandiego.geodatabase").arg(defaultDataPath());
```

Comment:
the geodatabase and route task don't need the view so could be moved to the c'tor